lib/ynab/models/account_type.rb

```diff
@@ -41,8 +41,12 @@ def self.build_from_hash(value)
     # @param [String] The enum value in the form of the string
     # @return [String] The enum value
     def build_from_hash(value)
-      return value if AccountType.all_vars.include?(value)
-      raise "Invalid ENUM value #{value} for class #AccountType"
+      if AccountType.all_vars.include?(value)
+        return value
+      else
+        return nil
+      end
```

Comment:
I'd expect these changes not to stick and a regression to happen as soon as the API Ruby code is regenerated from the OpenAPI spec file (for example using `rake generate`).

Is there a mechanism in place to prevent this kind of accidental regression to happen?

Note the comment at the top of most files:

https://github.com/ynab/ynab-sdk-ruby/blob/f3266625a91cc98dbcf6ecaf86af25b0630e4164/lib/ynab/models/account_type.rb#L6

Reply:
Yes - that template should ensure the changes stay intact on next generation.